### ISSUE TITLE
Add option for computer to only see certain files over MTP

### ISSUE THREAD

**Issue by FieryBinary** [GrapheneOS/os-issue-tracker#1383]
Currently, when transferring files over MTP, the computer can see all the files on the phone. This is insecure. If you want to transfer files, you must trust that the computer you connect to is not compromised or malicious, which is especially bad if you are connecting to someone else's computer or a public computer.

This could be solved by only allowing the computer access to certain directories, like Storage Scopes. When connecting, thYe user would be prompted to "configure Transfer Scopes" and select a directory that the computer can access. The computer would be none the wiser, and the user can securely transfer files without letting the computer see everything on the phone.

This would be fairly easy to do since it just applies Storage Scopes to MTP, but it would be a very useful feature.

**Comment by thestinger** [comment]
You can also transfer the files with a flash drive instead. We recommend this already:

https://grapheneos.org/faq#file-transfer

**Comment by FieryBinary** [comment]
@thestinger 

That's true, but connecting the phone and computer is simpler and quicker. Also, the user might not have a flash drive, or have a non-FAT32 drive that he cannot reformat.

**Comment by thestinger** [comment]
exFAT will be supported too for Pixel 6 and beyond with Android 13.

**Comment by FieryBinary** [comment]
@thestinger 

The exact filesystem doesn't matter - if the drive filesystem is not one of the Android-supported filesystems, the user is unable to transfer files to it. There are thousands of filesystems out there, and it's not feasible to support even the common ones like NTFS, BTRFS, HFS+, ext4...you get the point. It is far easier to connect to the computer, which currently poses security risks.

**Comment by thestinger** [comment]
exFAT / FAT32 are the only sensible filesystems for portable external drives though and I don't think it's that much harder to use a USB drive.

**Comment by thestinger** [comment]
exFAT / FAT32 are the only sensible filesystems for portable (as in cross-platform) external drives though and I don't think it's that much harder to use a USB drive.

**Comment by FieryBinary** [comment]
Regardless of how sensible other filesystems are, people do use other filesystems on flash drives. They often don't know or consider portability when formatting the drive, and once the drive is in use, are unable to format it since there is data on it.

The user might not have a flash drive on him. He could be out somewhere (an internet cafe, a library, a hotel...anywhere) without a flash drive in his pocket and cannot safely transfer files. Or the drive doesn't have enough space for a file - I cannot transfer a 2GB video if I only have 1 GB left. There are many possible things that prevent a user from using a flash drive and make connecting to the computer the only reliable way to transfer files.

**Comment by FieryBinary** [comment]
Any update on this?

**Comment by r3g-5z** [comment]
No, it's not high priority and anyone is welcome to work on this for themselves and submit a PR.
